CHANGELOG.md

```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Renamed `libcnb-data`'s `Buildpackage` to `PackageDescriptor`. This required changes in many other names across multiple libcnb.rs crates for consistency. See the PR diff for details, but renames should be straightforward and unsurprising. ([#656](https://github.com/heroku/libcnb.rs/pull/656))
```

Comment:
I kept it brief instead of iterating all the changed names. To me, there isn't much value in listing all renames when the renaming schema should be obvious. I can see an argument for listing all, so happy to discuss here.